lib/NetworkManager.js

```diff
@@ -186,9 +188,11 @@ class NetworkManager extends EventEmitter {
    * @param {string} url
    * @param {string} resourceType
    * @param {!Object} requestPayload
+   * @param {string} frameId
    */
-  _handleRequestStart(requestId, interceptionId, url, resourceType, requestPayload) {
-    const request = new Request(this._client, requestId, interceptionId, this._userRequestInterceptionEnabled, url, resourceType, requestPayload);
+  _handleRequestStart(requestId, interceptionId, url, resourceType, requestPayload, frameId) {
+    const frame = this._frameManager._frames.get(frameId);
```

Comment:
nit: can you please add a `FrameManager.frame(frameId)` method that would fetch frame for the given frameId. This is to avoid reaching into frameManager internals.

Reply:
@aslushnikov 
How about adding a underscore prefix and making a `FrameManager._frame(frameId)` method?
When you make a method without underscore, it should be documented on API docs.
It's not strictly a private method I know, but puppeteer's users shouldn't have to know about frameId in my opinion.
Or else, maybe expose `frame.id()` method.

Reply:
Ooops, never mind! `FrameManager` wasn't on the doc from the first place. I'll do that very quick!

Reply:
Done!
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/82215f41627fd14f619b1ce5c33c91e67c1971a3